UnityProject/Assets/Scripts/Items/Dice/RollDie.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Random = UnityEngine.Random;
+
+/// <summary>
+/// An extendable MonoBehaviour component to handle generic die rolling.
+/// </summary>
+public class RollDie : MonoBehaviour, IExaminable, ICheckedInteractable<HandActivate>
+{
+	[Tooltip("The amount of sides this die has.")]
+	public int sides = 6;
+	[Tooltip("Whether the die can be rigged to give flawed results.")]
+	public bool riggable = true;
```

Comment:
Public properties should be capitalized.

Reply:
These are fields not properties. So fields should be camelCase and properties should be PascalCase